### PR TITLE
Add debug logging

### PR DESCRIPTION
**Pull request by pfrazee**
@mafintosh This will need some careful review by you, to make sure I've correctly logged everything. Please add/remove/change however you see fit. (I have one debug() commented out with a TODO because I dont know how to interpret the situation; that'll definitely need to be fixed.)

Can test with `DEBUG=dns-discovery npm test`.

### CODE REVIEW

**Comment by pfrazee** [comment]
bump

**Comment by mafintosh** [comment]
merged the other two PRs. this one is a little bigger so gonna take some time later today and look through it

**Comment by pfrazee** [comment]
sounds good

File: index.js [hunk 1]
```diff
@@ -65,14 +66,17 @@ function DNSDiscovery (opts) {
   }
 
   function onerror (err) {
+    debug('Error', err)
     self.emit('error', err)
   }
 
   function onmulticastquery (message, rinfo) {
+    debug('MDNS query', rinfo.address + ':' + rinfo.port, message.questions.length, 'questions', message.answers.length, 'answers', message.additionals.length, 'additionals')
```
Comment (mafintosh) [inline index.js]: this line is a bit long. can we make it shorter / split it up?

File: index.js [hunk 2]
```diff
@@ -227,6 +249,8 @@ DNSDiscovery.prototype._onquestion = function (query, port, host, answers, multi
   var domain = parseDomain(query.name)
 
   if (query.type === 'TXT' && domain === query.name) {
+    // TODO wait, what is this? is it an unsubscribe? that's the only codepath in _onanswer this seems to match
+    // debug('Replying announce via TXT to', host + ':' + port)
```
Comment (mafintosh) [inline index.js]: This is a debugging thing. You can use tools like `dig` to query for a text record the get information about a domain. Useful to sanity check things in production.  Could you remove the comment or update it?

**Comment by mafintosh** [comment]
@pfrazee Looking good! One minor comment. Thanks for adding all the debug information btw. I know I'm terrible at doing that.

**Comment by pfrazee** [comment]
Ok should be good to go

**Comment by mafintosh** [comment]
very good. 5.5.0

**Comment by pfrazee** [comment]
thanks!

**Pull request merged**